bpf/lib/policy.h

```diff
@@ -123,6 +123,47 @@ __policy_can_access(const void *map, struct __ctx_buff *ctx, __u32 localID,
 	}
 #endif /* ALLOW_ICMP_FRAG_NEEDED */
 
+	/* When the ICMP or ICMPv6 packet come, set its type to the key.dport*/
+	if (proto == IPPROTO_ICMP) {
+		void *data, *data_end;
+		struct iphdr *ip4;
+		__u8 icmp_type;
+		__u32 off;
+
+		if (!revalidate_data(ctx, &data, &data_end, &ip4))
+			return DROP_INVALID;
+
+		off = ((void *)ip4 - data) + ipv4_hdrlen(ip4);
+		if (ctx_load_bytes(ctx, off, &icmp_type, sizeof(icmp_type)) < 0)
+			return DROP_INVALID;
+
+		/* convert from unsigned char to unsigned short considering byteorder */
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+		key.dport = (__u16)icmp_type;
+#else
+		key.dport = (__u16)(icmp_type << 8);
+#endif
```

Comment:
I'd rather we make such checks in userspace, mostly because:
1. If we can move complexity to userspace without cost, we should.
2. We will likely want to add support for ICMP codes later on and in that case the type needs to be on the 8 upper bits.

Reply:
I agree with you :)
However I don't know how to check it in the userspace.
Because we get `icmp_type` from `ctx_load_bytes()` and it's always 8bit, so we have to expand it to 16bits in bpf.
Do you have any idea?

Reply:
Actually, I don't think we support big-endian architectures in the agent at all: https://github.com/cilium/cilium/blob/v1.10.1/pkg/byteorder/byteorder.go#L70-L79. So it's probably fine to only support little-endian here.

Reply:
Sure, I'll remove big-endian case.

Reply:
Just a heads up, the statement about not supporting big-endian is incorrect. The swap is done inside the swap functions:

https://github.com/cilium/cilium/blob/e6f34c3c98fe2e247fde581746e552d8cb18c33c/pkg/byteorder/byteorder.go#L42